jax/lib/xla_bridge.py

```diff
@@ -179,19 +179,21 @@ def device_count(backend=None):
   return int(get_backend(backend).device_count())
 
 
-def local_device_count(backend=None):
+def local_device_count(backend: str =None):
   """Returns the number of devices on this host."""
   return int(get_backend(backend).local_device_count())
 
 
-def devices(backend=None):
-  """Returns a list of all devices.
+def devices(backend: str = None):
+  """Returns a list of all devices for a given backend.
 
-  Each device is represented by a subclass of Device (e.g. CpuDevice,
-  GpuDevice). The length of the returned list is equal to
-  ``device_count()``. Local devices can be identified by comparing
+  Each device is represented by a subclass of ``Device`` (e.g. ``CpuDevice``,
+  ``GpuDevice``). The length of the returned list is equal to
+  ``device_count(backend)``. Local devices can be identified by comparing
   ``Device.host_id`` to ``host_id()``.
 
+  If ``backend`` is ``None``, returns all the devices from the default backend.
```

Comment:
I think it is useful to explain that backends have a priority

Reply:
Done.